Products/CMFPlone/browser/icon.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+from OFS.Image import File
+from plone.registry.interfaces import IRegistry
+from Products.CMFCore.interfaces import ISiteRoot
+from zope.component import adapter
+from zope.component import getUtility
+from zope.component.hooks import getSite
+from zope.interface import implementer
+from zope.interface import Interface
+from zope.location.interfaces import LocationError
+from zope.traversing.interfaces import ITraversable
+from zope.publisher.interfaces import IPublishTraverse
+
+
+@implementer(IPublishTraverse)
+class IconView:
+
+    prefix = 'plone.staticresources.icon.'
+
+    def publishTraverse(self, request, name):
+        request['TraversalRequestNameStack'] = []
+        return self
+
+    def __init__(self, context, request):
+        super(IconView, self).__init__(context, request)
+        if len(self.request.path) == 1:
+            self.icon = request.path[0]
+
+    def __call__(self, icon=None):
+        if icon:
+            self.icon = icon
+        name = self.icon
+        url = getSite().absolute_url() + '/' + self.get_icon(name)
+        return url
+
+    def get_icon(self, name):
```

Comment:
i dont like the Java'esc `get_icon`. I would name it `registered_icon` or `icon_from_registry` (latter is what it is).